### PR TITLE
Quantile type conversion

### PR DESCRIPTION
**Pull request by jmalkin**
i think this is ready

### CODE REVIEW

File: quantiles/include/quantiles_sketch_impl.hpp [hunk 1]
```diff
@@ -138,6 +138,66 @@ is_sorted_(is_sorted)
...
+                "Copy constructor across types requires std::is_convertible or std::is_constructible");
+
+  base_buffer_.reserve(2 * std::min(quantiles_constants::MIN_K, k_));
+
+  if (!other.is_empty()) {
+    min_value_ = new (allocator_.allocate(1)) T(other.get_min_value());
+    max_value_ = new (allocator_.allocate(1)) T(other.get_max_value());
+
+    // reserve space in levels
+    uint8_t num_levels = compute_levels_needed(k_, n_);
```
Comment (AlexanderSaydakov) [inline quantiles/include/quantiles_sketch_impl.hpp:165]: should be const

File: quantiles/include/quantiles_sketch_impl.hpp [hunk 2]
```diff
@@ -138,6 +138,66 @@ is_sorted_(is_sorted)
...
+    levels_.reserve(num_levels);
+    for (int i = 0; i < num_levels; ++i) {
+      Level level(allocator);
+      level.reserve(k_);
+      levels_.push_back(std::move(level));
+    }
+
+    // iterate through points, assigning to the correct level as needed
+    for (auto pair : other) {
+      uint64_t wt = pair.second;
```
Comment (AlexanderSaydakov) [inline quantiles/include/quantiles_sketch_impl.hpp:175]: const

File: quantiles/include/quantiles_sketch_impl.hpp (hunk 1 above)
Comment (jmalkin) [reply quantiles/include/quantiles_sketch_impl.hpp]: fixed

File: quantiles/include/quantiles_sketch_impl.hpp (hunk 2 above)
Comment (jmalkin) [reply quantiles/include/quantiles_sketch_impl.hpp]: done

File: quantiles/include/quantiles_sketch_impl.hpp [hunk 3]
```diff
@@ -138,6 +138,66 @@ is_sorted_(is_sorted)
...
+    }
+
+    // iterate through points, assigning to the correct level as needed
+    for (auto pair : other) {
+      const uint64_t wt = pair.second;
+      if (wt == 1) {
+        base_buffer_.push_back(pair.first);
+        // resize where needed as if adding points via update()
+        if (base_buffer_.size() + 1 > base_buffer_.capacity()) {
+          size_t new_size = std::max(std::min(static_cast<size_t>(2 * k_), 2 * base_buffer_.size()), static_cast<size_t>(1));
```
Comment (AlexanderSaydakov) [inline quantiles/include/quantiles_sketch_impl.hpp:180]: and this one should be const

**Review by AlexanderSaydakov — APPROVED** [review]
👍

**Pull request merged**